tests/clients/test_job.py

```diff
@@ -84,7 +76,53 @@
 
 JOB_SUMMARY = JobSummary(id=JOB_ID, metadata=JOB_METADATA)
 JOB_SUMMARY_BODY = {"jobId": str(JOB_ID), "meta": JOB_METADATA_BODY}
-
+INSTANCE_SIZE = InstanceSize(milli_cpus=MILLI_CPUS, memory_mb=MEMORY_MB)
+INSTANCE_SIZE_BODY = {"milliCpus": MILLI_CPUS, "memoryMb": MEMORY_MB}
+JOB_PARAMETER = JobParameter(
+    name="job name",
+    type=ParameterType.TEXT,
+    default="default parameter value",
+    required=True,
+)
+JOB_PARAMETER_BODY = {
+    "name": JOB_PARAMETER.name,
+    "type": "text",
+    "default": JOB_PARAMETER.default,
+    "required": True,
+}
+JOB_COMMAND = JobCommand(name="python myscript.py", parameters=[JOB_PARAMETER])
+JOB_COMMAND_BODY = {
+    "name": JOB_COMMAND.name,
+    "parameters": [JOB_PARAMETER_BODY],
+}
+JOB_DEFINITION = JobDefinition(
+    working_dir="/project/subdir/",
+    command=JOB_COMMAND,
+    image_type=ImageType.PYTHON,
+    conda_environment="Python3",
+    environment_ids=[str(ENVIRONMENT_ID)],
+    instance_size_type="custom",
+    instance_size=INSTANCE_SIZE,
+    max_runtime_seconds=MAX_RUNTIME_SECONDS,
+)
+JOB_DEFINITION_BODY = {
```

Comment:
As described in my comment on the implementation above, I think we need to also test the deserialisation of a non-custom image type job definition. Also, we should test the deserialisation of an R image type job definition (where the conda_environment will be `None`. As with the invalid test cases comment below, we probably want to make this a parametrized test.

Reply:
Ok, I'll get to it - by parametrized test do you mean using the stuff described in here? http://doc.pytest.org/en/latest/example/parametrize.html

Reply:
Confusingly, that's the 'advanced' parameterisation docs, I think you want more:
http://doc.pytest.org/en/latest/parametrize.html#parametrize-basics

Or just see the existing use of `pytest.mark.parametrize` in `test_job.py`.